File: src/main/java/org/goafabric/core/data/logic/mapper/PractitionerMapper.java
package org.goafabric.core.data.logic.mapper;

import org.goafabric.core.data.controller.vo.Practitioner;
import org.goafabric.core.data.repository.entity.PractitionerEo;
import org.mapstruct.Mapper;
import org.mapstruct.ReportingPolicy;

import java.util.List;


@Mapper(componentModel = "jakarta", unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface PractitionerMapper extends WorkaroundMapper {
    Practitioner map(PractitionerEo value);

    PractitionerEo map(Practitioner value);

    List<Practitioner> map(List<PractitionerEo> value);

}

File: src/main/kotlin/org/goafabric/core/organization/logic/mapper/PractitionerMapper.kt
package org.goafabric.core.organization.logic.mapper

import org.goafabric.core.organization.controller.dto.Practitioner
import org.goafabric.core.organization.persistence.entity.PractitionerEo
import org.mapstruct.Mapper
import org.mapstruct.ReportingPolicy

@Mapper(componentModel = "cdi", unmappedTargetPolicy = ReportingPolicy.IGNORE, uses = [AddressMapper::class])
interface PractitionerMapper {
    fun map(value: PractitionerEo): Practitioner
    fun map(value: Practitioner): PractitionerEo
    fun map(values: List<PractitionerEo>): List<Practitioner>
}
